src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -22,8 +13,29 @@ public TaskController(DatastoreService datastore) {
     this.datastore = datastore;
   }
 
-  public HashSet<TaskData> getTasks(int quantity, String sortBy, String sortDirection) {
-    HashSet<TaskData> tasks = new HashSet<>();
+  public TaskData getTaskByID(long taskID) {
+    Query query = new Query("Task").addFilter("taskID", FilterOperator.EQUAL, taskID);
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asIterable().iterator().next();
+    TaskData task = new TaskData(entity);
+    return task;
+  }
+
+  public ArrayList<TaskData> getSubtasks(TaskData task) {
```

Comment:
This method and `getTasks` are mostly the same, just with different query options and filters.  Maybe look into extracting the common pieces into a `private` util method.